src/proc_diskstats.c

```diff
@@ -523,6 +669,40 @@ int do_proc_diskstats(int update_every, unsigned long long dt) {
 		}
 
 		// --------------------------------------------------------------------
+
+		if(ddo_space) {
+			if(!mount_point) {
+				if(ddo_space != CONFIG_ONDEMAND_ONDEMAND) {
+					error("Cannot find space usage for disk %s. It does not have a mount point.", family);
+				}
+			} else {
+				if (statvfs(family, &buff_statvfs) < 0) {
+					error("Failed checking disk space usage of %s", family);
+				} else {
+					space_avail = buff_statvfs.f_bavail * buff_statvfs.f_bsize;
+					space_avail_root = (buff_statvfs.f_bfree - buff_statvfs.f_bavail) * buff_statvfs.f_bsize;
+					space_used = (buff_statvfs.f_blocks - buff_statvfs.f_bfree) * buff_statvfs.f_bsize;
+
+					st = rrdset_find_bytype("disk_space", disk);
+					if(!st) {
+						st = rrdset_create("disk_space", disk, NULL, family, "disk.space", "Disk Space Usage", "GB", 2023, update_every, RRDSET_TYPE_STACKED);
+						st->isdetail = 1;
+
+						rrddim_add(st, "avail", NULL, 1, 1000*1000*1000, RRDDIM_ABSOLUTE);
+						rrddim_add(st, "reserved for root", NULL, 1, 1000*1000*1000, RRDDIM_ABSOLUTE);
```

Comment:
I should have said this before:

the second argument is best not to contain spaces.
the third (shown as NULL) is the name that will appear in the dashboard.

so, you can define this as:

```
rrddim_add(st, "reserved_for_root", "reserved for root", 1, 1000*1000*1000, RRDDIM_ABSOLUTE);
```

Later, at `rrdim_set()` you should use the id: `reserved_for_root`